### ISSUE TITLE
Issue with Training and testing

### ISSUE THREAD

**Issue by avisin02** [yongxuUSTC/sednn#10]
Hello Yong,
I am very much impressed with work. I am trying to use the code on my data. My setup is as follows:
1. Right now I am using Python 3.6 on windows 10.
2. I have implemented runme.sh in a python code.
4. For training I am using 1350 sentences and 8 types of noises of duration same as combined duration of sentences.
5. For testing I am using 150 sentences with some noise same as training noise and some others.
Now, the issue is that the training is less but not close to 0 and testing loss is way more than expected and enhanced wav files have no sign of speech in any noise condition.
What should be done in this case? I am also attaching the plot of training stat.
![plot](https://user-images.githubusercontent.com/39087945/39755651-53ddafbc-52e4-11e8-98a8-fe59700eb922.png)

Regards,
Ajay

**Comment by yongxuUSTC** [comment]
Hi Ajay,

Have you tried on Linux?
If you found that your test loss was quite large (over-fitting), I suggest you can try some simple experiments first, e.g., clean-to-clean, or with only one type noise in the training and the same noise type in the testing. This can help you to test your code.

BTW, have you test one utterance of your training set? Is the reconstructed wav audible or not?

Best regards,
yong
________________________________
--------------------------------------------------------
Dr. Yong XU
https://sites.google.com/view/xuyong/home

**Comment by avisin02** [comment]
Hi Yong,

I haven't checked on Linux and there some minor issues between python
version 2.x and 3.x.


1. For large data the samples are not over fitting but test as well as the
training loss is large.
    Earlier the wav files were not audible.

2. With 1 sentence and 1 noise type for both training and testing
    training loss = 0.080497
    wav file is audible but distorted.


3. Now, I have checked with 1 sentence with 2 noise types for training
    the training loss is 0.034137
    The wav file are audible but distorted.

Please suggest ways to improve this and I will also try it in Linux but
right now I am working windows.

Regards,
Ajay

**Comment by yongxuUSTC** [comment]
Hi Ajay,

I suggest "1000 sentences without any noise" which means "clean to clean mapping". You can firstly test this one. To see whether you can recover your wav for training set or testing set.

Best regards,
yong
________________________________
--------------------------------------------------------
Dr. Yong XU
https://sites.google.com/view/xuyong/home

**Comment by bailiangze** [comment]
do you solve your problem , I miss the same problem

**Comment by qiuqiangkong** [comment]
Hi Ajay,


    Did you check the sampling rate of the speech and noise? They should all be resampled to 16 kHz.


    Did you try with TIMIT dataset with the noise dataset mentioned in the code? If it works then it might be the problem of the dataset you use. If it still not working it might be some problem of the configuration of the code on the windows machine.


Best wishes,


Qiuqiang

________________________________

**Comment by bailiangze** [comment]
thank you for your reply
I use the orign wav file in mini data。I use this in window7.
![figure_1-1](https://user-images.githubusercontent.com/42959668/50499696-a87a6600-0a86-11e9-8911-8f4a3513acb6.png)
![test](https://user-images.githubusercontent.com/42959668/50499732-eecfc500-0a86-11e9-87e4-44f728400f70.png)

**Comment by bailiangze** [comment]
when I run the code in mixture2clean .it occurs a probelm。I changed the code from n_pad = (n_concat - 1) / 2 to int（n_pad = (n_concat - 1) / 2） .I don't think this change will affect the code.

**Comment by qiuqiangkong** [comment]
Hi,


    If mini data is used, the loss curve may behave what you plot. Because there are too little data for training so will not generalize to the testing data.


Best wishes,


Qiuqiang

________________________________

**Comment by qiuqiangkong** [comment]
Hi,


    If mini data is used for training, you may get enhanced speech but they are not good. But at least you should be able to recognize the speech. Using more data will lead to a better result.


Best wishes,


Qiuqiang

________________________________

**Comment by bailiangze** [comment]
Thank you 。Is  the RBM code  include in this code？
